client/state/data-layer/wpcom/comments/index.js

```diff
@@ -156,6 +160,35 @@ export const announceFailure = ( { dispatch, getState }, { siteId, postId } ) =>
 	dispatch( errorNotice( error ) );
 };
 
+export const deleteComment = ( { dispatch }, action ) => {
+	const { siteId, commentId } = action;
+	dispatch(
+		http( {
+			method: 'POST',
+			apiVersion: '1.1',
+			path: `/sites/${ siteId }/comments/${ commentId }/delete`,
+			onSuccess: {
+				...action,
+				type: COMMENTS_DELETE_SUCCESS,
+			},
+			onFailure: {
+				...action,
+				type: COMMENTS_DELETE_FAILURE,
+			},
+		} )
+	);
+};
+
+export const removeCommentFromState = ( { dispatch }, action ) => {
+	const { siteId, postId, commentId } = action;
+	dispatch( removeComment( siteId, postId, commentId ) );
```

Comment:
> Currently COMMENTS_DELETE doesn't pass through the reducer, while COMMENTS_REMOVE does.
I could add _DELETE as a fall through of _REMOVE?

If it hits your handler in the data layer it will hit the reducer after the handlers have finished. What makes you think it's not passing through the reducer? Is there a reducer listening to that action type?

> Also: I've noticed that in both the success and failure functions, the dispatches don't work.
The functions are indeed called (a console log in there works fine) removeComment and errorNotice are never fired (at least according to the Redux Dev Tools).

If this truly is the case then there's probably something else wrong. How do you know they aren't firing? Can you look at the state before and after running this action to confirm instead of just relying on the dev tools? they are known to be misleading at times.

```
// in dev tools
state // returns state
state.comments // returns comments state
```